scipy/stats/tests/test_stats.py

```diff
@@ -5700,3 +5700,24 @@ def test_dist_perm(self):
                                                                random_state=1)
         assert_approx_equal(stat_dist, 0.163, significant=1)
         assert_approx_equal(pvalue_dist, 0.001, significant=1)
+
+class TestConfInt(object):
+    """ Test the computation of non-parametric
+    confidence intervals for quantiles
+    """
+    X = array([2, 8, 3, 6, 4, 1, 5, 9, 7], float)
+    
+    def test_index_equal_value(self):
+        assert_equal(stats.confint_quantile(X, 0.5, 0.9), (3.0, 7.0))
+        assert_equal(stats.confint_quantile(X.shape[0], 0.5, 0.9), (2, 6))
+
+    def test_twosided(self):
+        assert_equal(stats.confint_quantile(X.shape[0], 0.5, 0.9, type='two-sided'), (1, 7))
+
+    def test_values(self):
+        N, q, c = 100, 0.75, 0.95
+        assert_equal(stats.confint_quantile(N, q, c), (67, 82))
+        N, q, c = 10, 0.75, 0.95
+        assert_equal(stats.confint_quantile(N, q, c), (4, None))
+        N, q, c = 20, 0.175, 0.95
+        assert_equal(stats.confint_quantile(N, q, c), (0, 6))
```

Comment:
you could parametrize over `N`, `q`, `c`  and the expected return tuples to reduce duplication, but I wouldn't bother doing that unless another reviewer also asks for it (and wait till after a proper stats reviewer checks :) )

Reply:
Sounds like a good idea to me, as I'd like to see more of these if they are coming from some "ground truth" (so I can confirm that `confint_quantile` is doing the same thing as some reference).